activesupport/lib/active_support/string_inquirer.rb

```diff
@@ -9,6 +9,34 @@ module ActiveSupport
   #
   #   Rails.env.production?
   class StringInquirer < String
+
+    def initialize(value, valid_values = [])
```

Comment:
Let's do it this way:

```ruby
def initialize(value, restricted_to = [])
  super(value)
  @restricted_to = restricted_to
end
```

And then just check `@restricted_to` in `respond_to_missing?` and `method_missing?`.